backend/src/core/errors/transactional.errors.ts

```diff
@@ -6,10 +6,11 @@ export class RateLimitError extends Error {
   }
 }
 
-export class InvalidMessageError extends Error {
-  constructor(
-    message = 'Message template is invalid as it only contains invalid HTML tags'
-  ) {
+export const EMPTY_MESSAGE_ERROR =
+  'Message is empty after removing invalid HTML tags'
+
+export class EmptyMessageError extends Error {
```

Comment:
This name change is misleading cos the original message is not `empty` it's `invalid`

Reply:
so this error is only thrown if the message becomes empty after removing the invalid HTML tags

normally, if there are invalid HTML tags, they will just be automatically removed and this error won't be thrown. that's why I renamed it

(you can see that's what the original message was getting at `message = 'Message template is invalid as it only contains invalid HTML tags'` → it becomes empty after the invalid HTML tags are removed)

Reply:
> so this error is only thrown if the message becomes empty after removing the invalid HTML tags
> 
> normally, if there are invalid HTML tags, they will just be automatically removed and this error won't be thrown. that's why I renamed it
> 
> (you can see that's what the original message was getting at `message = 'Message template is invalid as it only contains invalid HTML tags'` → it becomes empty after the invalid HTML tags are removed)
> 
> to be exact, it could be either the subject or the body that is empty, but if at least either one is empty, the error will be thrown

Hmm but I feel that we don't need that kinda granularity here. `Invalid` would be more future-proof in case we add more rules to this validation

Reply:
Hmm but I must say that in the beginning, when I was trying to cause this error before I looked at the code in detail, I assumed that the error would be thrown if invalid HTML tags were used or if the entire message was wrapped around invalid HTML tags.

Then after poking further, I realise that our code removed the HTML tags directly and only throws the error in the edge case that this removal causes the field to be empty.

I would prefer to be a bit more specific, but it's no big deal either way. Will defer to your decision.

Reply:
> Hmm but I must say that in the beginning, when I was trying to cause this error before I looked at the code in detail, I assumed that the error would be thrown if invalid HTML tags were used or if the entire message was wrapped around invalid HTML tags.
> 
> Then after poking further, I realise that our code removed the HTML tags directly and only throws the error in the edge case that this removal causes the field to be empty.
> 
> I would prefer to be a bit more specific, but it's no big deal either way. Will defer to your decision.

I think that `Empty` is actually more misleading, as the original message is **not** empty

Reply:
Have renamed it `MessageError` — feel like it's more consistent with the `TemplateError` [thrown elsewhere](https://github.com/opengovsg/postmangovsg/blob/08da008200a75721bf0a23283f65409ad2e89d2d/backend/src/email/services/email-template.service.ts#L180-L184). 

Have added the test cases too.